crates/core/src/client_events/mod.rs

```diff
@@ -1079,6 +1082,78 @@ async fn process_open_request(
             ClientRequest::NodeQueries(query) => {
                 tracing::debug!("Received node queries from user event: {:?}", query);
 
+                // Handle ProximityCacheInfo directly without creating a NodeEvent
+                if matches!(
+                    query,
+                    freenet_stdlib::client_api::NodeQuery::ProximityCacheInfo
+                ) {
+                    // Return proximity cache data
+                    let (my_cache_hashes, neighbor_cache_data) =
+                        proximity_cache.get_introspection_data().await;
+                    let stats = proximity_cache.get_stats().await;
+
+                    // Convert internal data to API types
+                    let my_cache = my_cache_hashes
+                        .into_iter()
+                        .map(|hash| {
+                            freenet_stdlib::client_api::ContractCacheEntry {
+                                contract_key: format!("hash_{:08x}", hash), // Show hash representation
+                                cache_hash: hash,
+                                cached_since: std::time::SystemTime::now()
+                                    .duration_since(std::time::UNIX_EPOCH)
+                                    .unwrap_or_default()
+                                    .as_secs(),
+                            }
+                        })
+                        .collect();
+
+                    let neighbor_caches: Vec<_> = neighbor_cache_data
+                        .into_iter()
+                        .map(|(peer_id, contracts)| {
+                            freenet_stdlib::client_api::NeighborCacheInfo {
+                                peer_id,
+                                known_contracts: contracts,
+                                last_update: std::time::SystemTime::now()
+                                    .duration_since(std::time::UNIX_EPOCH)
+                                    .unwrap_or_default()
+                                    .as_secs(),
+                                update_count: 1, // Placeholder
+                            }
+                        })
+                        .collect();
+
+                    // Calculate average neighbor cache size
+                    let total_neighbors = neighbor_caches.len() as f64;
+                    let total_contracts: f64 = neighbor_caches
+                        .iter()
+                        .map(|n| n.known_contracts.len() as f64)
+                        .sum();
+                    let avg_cache_size = if total_neighbors > 0.0 {
+                        total_contracts / total_neighbors
+                    } else {
+                        0.0
+                    };
+
+                    let proximity_info = freenet_stdlib::client_api::ProximityCacheInfo {
```

Comment:
why is any of this logic going throught the ws client events handling even? shouldn't this belong at the p2p connection layer?